android/KMEA/app/src/main/java/com/tavultesoft/kmea/cloud/impl/CloudLexicalModelMetaDataDownloadCallback.java

```diff
@@ -148,7 +148,7 @@ private void processCloudResults(Context aContext, List<MetaDataResult> aMetaDat
             KMLog.LogException(TAG, "Error parsing lexical model from api.keyman.com. ", e);
           }
         } else {
-          BaseActivity.makeToast(aContext, R.string.no_associated_model, Toast.LENGTH_SHORT);
+          // Don't make a Toast notification about nothing to download
```

Comment:
Suggest we just delete the `else` block altogether -- we have commit history for the 'why'.

In fact, would be better to invert the test on line 134 to `continue`? Perhaps we should do a little cleanup? I would do something like this:

```java
  private void processCloudResults(Context aContext, List<MetaDataResult> aMetaDataResult) {
    for(MetaDataResult _r:aMetaDataResult) {
      if (_r.returnjson.target== CloudApiTypes.ApiTarget.Keyboard) {
        //handleKeyboardMetaData(_r);
      }
      if(_r.returnjson.target== CloudApiTypes.ApiTarget.KeyboardLexicalModels) {
        processCloudResultForModel(_r.returnjson.jsonArray);
      }
    }
  }

  private void processCloudResultForModel(JSONArray lmData) {
    if (lmData == null || lmData.length() == 0) {
      KMLog.LogError(TAG, "Error in lexical model metadata from api.keyman.com - zero or null");
      return;
    }

    try {
      JSONObject modelInfo = lmData.getJSONObject(0);
      if (!modelInfo.has("packageFilename") || !modelInfo.has("id")) {
        KMLog.LogError(TAG, "Error in lexical model metadata from api.keyman.com - missing metadata");
        return;
      }

      String _modelID = modelInfo.getString("id");
      ArrayList<CloudApiTypes.CloudApiParam> urls = new ArrayList<>();
      urls.add(new CloudApiTypes.CloudApiParam(
        CloudApiTypes.ApiTarget.LexicalModelPackage,
        modelInfo.getString("packageFilename")));
      _r.additionalDownloadid = CloudLexicalPackageDownloadCallback.createDownloadId(_modelID);
      _r.additionalDownloads= urls;
    } catch (JSONException e) {
      KMLog.LogException(TAG, "Error in lexical model metadata from api.keyman.com. ", e);
    }
  }
```

Not sure if LogError exists but if it doesn't it should. Logging something going wrong that we are catching to sentry -- we want to know if lm data from the server is bogus.

Ironically, I would suggest adding toasts to LogError and LogException if they are not there already, so the user knows something has gone wrong and can report the error to us (assuming they remember the message at all!)

Reply:
> Ironically, I would suggest adding toasts to LogError and LogException if they are not there already, so the user knows something has gone wrong and can report the error to us (assuming they remember the message at all!)

I split that to a separate issue, since that may need more testing to see if adding Toasts to KMLog make the app too noisy. (Hopefully not)